---
title: "Wyniki"
output: html_document
date: "2025-02-04"
---


## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:
```{r setup, include=FALSE}
# , include=FALSE oznacza, że ten fragment kodu nie zostanie wygenerowany w raporcie
knitr::opts_chunk$set(echo = TRUE)
setwd("~/Documents/Studia/GenPop")
library(ggplot2)
library(tidyr)
library(dplyr)

```



```{r wczytaj plik VCF z chCADD scores}
vcf_1 <- read.table("/home/frankie/Documents/Studia/GenPop/Lab7/vcf_C_pyg_26_forward_indelRm_intersect_annotated_SNP_is_alt.bed")[,c(1,3,6,20)]

vcf_2 <- read.table("/home/frankie/Documents/Studia/GenPop/Lab7/vcf_C_ruf_09_forward_indelRm_intersect_annotated_SNP_is_alt.bed")[,c(1,3,6,20)]

colnames(vcf_1) <- c("chr", "pos", "chCADD", "FORMAT")
colnames(vcf_2) <- c("chr", "pos", "chCADD", "FORMAT")

vcf_1$gat <- "C_pyg"
vcf_2$gat <- "C_ruf"

vcf <- rbind(vcf_1, vcf_2)
# policz warianty
n1_variants <- nrow(vcf_1)

# wyciągnij informacje o homozygotach/heterozygotach. Użyj funkcji separate z paczki tidyr. 
vcf_1 %>%
  separate(FORMAT, into = "genotype", sep = ":")

# jeśli chcesz pozbyć się ostrzeżenia, dodaj paramter extra:
vcf_1 %>%
  separate(FORMAT, into = "genotype", sep = ":", extra = "drop") 

# zmiana 0/0 na hom_ref etc. 
vcf_1 <- vcf_1 %>%
  separate(FORMAT, into = "genotype", sep = ":", extra = "drop") %>%
  mutate(genotype = ifelse(genotype == "0/1", "HET", 
                           ifelse(genotype == "1/1", "HOM_ALT", 
                                  ifelse(genotype == "0/0", "HOM_REF", "error"))))


# policz warianty
n2_variants <- nrow(vcf_2)

# wyciągnij informacje o homozygotach/heterozygotach. Użyj funkcji separate z paczki tidyr. 
vcf_2 %>%
  separate(FORMAT, into = "genotype", sep = ":")

# jeśli chcesz pozbyć się ostrzeżenia, dodaj paramter extra:
vcf_2 %>%
  separate(FORMAT, into = "genotype", sep = ":", extra = "drop") 

# zmiana 0/0 na hom_ref etc. 
vcf_2 <- vcf_2 %>%
  separate(FORMAT, into = "genotype", sep = ":", extra = "drop") %>%
  mutate(genotype = ifelse(genotype == "0/1", "HET", 
                           ifelse(genotype == "1/1", "HOM_ALT", 
                                  ifelse(genotype == "0/0", "HOM_REF", "error"))))

# policz heterozygoty i homozygoty:
table(vcf_1$genotype)
table(vcf_2$genotype)

# policz ponownie:
vcf_1 %>%
  group_by(genotype) %>%
  tally()

n1_HET <- vcf_1 %>%
  group_by(genotype) %>%
  tally() %>%
  filter(genotype == "HET") %>%
  pull()

n1_HOM_ALT <- vcf_1 %>%
  group_by(genotype) %>%
  tally() %>%
  filter(genotype == "HOM_ALT") %>%
  pull()

# policz ponownie:
vcf_2 %>%
  group_by(genotype) %>%
  tally()

n2_HET <- vcf_2 %>%
  group_by(genotype) %>%
  tally() %>%
  filter(genotype == "HET") %>%
  pull()

n2_HOM_ALT <- vcf_2 %>%
  group_by(genotype) %>%
  tally() %>%
  filter(genotype == "HOM_ALT") %>%
  pull()

paste0("W danych zidentyfikowano ", n1_HET, " heterozygot i ", n1_HOM_ALT, " homozygot.")
paste0("W danych zidentyfikowano ", n2_HET, " heterozygot i ", n2_HOM_ALT, " homozygot.")

```

```{r wizualizacja}
ggplot(vcf_1, aes(pos, chCADD)) +
  geom_point() +
  ggtitle("chCADD, scaffold X, individual Y", 
          subtitle = paste0(n1_variants, " variants"))

ggplot(vcf_2, aes(pos, chCADD)) +
  geom_point() +
  ggtitle("chCADD, scaffold X, individual Y", 
          subtitle = paste0(n2_variants, " variants"))
# zwizualizuj chCADD wzdłuż scaffoldu z podziałem na genotyp.
```
```{r pyg_26 testowanie}
head(vcf_1)
ggplot(vcf_1, aes(chCADD)) +
  geom_histogram() +
  facet_wrap(~genotype)


vcf_1 %>%
  mutate(chCADD_log = log10(chCADD)) %>%
  ggplot(aes(chCADD_log)) +
  geom_histogram() +
  facet_wrap(~genotype)


vcf_1 %>%
  mutate(chCADD_log = log10(chCADD)) %>%
  ggplot(aes(chCADD_log)) +
  geom_density() +
  facet_wrap(~genotype)
```
```{r ruf_09 testowanie}
head(vcf_2)
ggplot(vcf_2, aes(chCADD)) +
  geom_histogram() +
  facet_wrap(~genotype)


vcf_2 %>%
  mutate(chCADD_log = log10(chCADD)) %>%
  ggplot(aes(chCADD_log)) +
  geom_histogram() +
  facet_wrap(~genotype)


vcf_2 %>%
  mutate(chCADD_log = log10(chCADD)) %>%
  ggplot(aes(chCADD_log)) +
  geom_density() +
  facet_wrap(~genotype)
```
```{r wykresy}
vcf_1 %>%
  ggplot(aes(chCADD)) +
  geom_density() +
  facet_wrap(~genotype)

vcf_1 %>%
  ggplot(aes(x = genotype, y = chCADD)) +
  geom_boxplot() +
  geom_jitter(width = .2) +
  ggtitle("C_pyg") +
  theme(plot.title = element_text(hjust = 0.5))

vcf_2 %>%
  ggplot(aes(x = genotype, y = chCADD)) +
  geom_boxplot() +
  geom_jitter(width = 0.2) +
  ggtitle("C_ruf") +
  theme(plot.title = element_text(hjust = 0.5))

wilcox.test(chCADD ~ genotype, data = vcf_1)

wilcox.test(chCADD ~ genotype, data = vcf_2)
```

```{r liczenie obciążeń}
# Współczynnik dominacji
h <- 0.1  

# Funkcja do obliczeń
calculate_loads <- function(data) {
  # Sumowanie wartości CADD dla HOM_ALT i HET
  L_hom <- sum(data$chCADD[data$genotype == "HOM_ALT"], na.rm=TRUE)
  L_het <- sum(data$chCADD[data$genotype == "HET"], na.rm=TRUE)
  
  # Obliczenia
  Genetic_Load <- L_hom + 0.5 * L_het
  Realized_Load <- L_hom + h * L_het
  
  return(data.frame(Genetic_Load, Realized_Load))
}

# Obliczenia dla C_pyg
load_pyg <- calculate_loads(vcf_1)
load_pyg

# Obliczenia dla C_ruf
load_ruf <- calculate_loads(vcf_2)
load_ruf

vcf_1$Genetic_Load <- load_pyg$Genetic_Load
vcf_1$Realized_Load <- load_pyg$Realized_Load

vcf_2$Genetic_Load <- load_ruf$Genetic_Load
vcf_2$Realized_Load <- load_ruf$Realized_Load

# Wykres dla C_pyg
ggplot(vcf_1, aes(x = pos, y = chCADD)) +
  geom_point(alpha = 0.5, color = "blue") +
  geom_smooth(method = "loess", color = "darkblue") +
  labs(title = "chCADD vs Position - C. pyg",
       x = "Genomic Position",
       y = "chCADD") +
  theme_minimal()

# Wykres dla C_ruf
ggplot(vcf_2, aes(x = pos, y = chCADD)) +
  geom_point(alpha = 0.5, color = "red") +
  geom_smooth(method = "loess", color = "darkred") +
  labs(title = "chCADD vs Position - C. ruf",
       x = "Genomic Position",
       y = "chCADD") +
  theme_minimal()

cor.test(vcf_1$pos, vcf_1$chCADD, method = "spearman")
cor.test(vcf_2$pos, vcf_2$chCADD, method = "spearman")

```
